core/src/services/gcs/core.rs

```diff
@@ -534,6 +561,54 @@ impl GcsCore {
         self.send(req).await
     }
 
+    pub async fn gcs_list_object_versions(
```

Comment:
The output is the same for GCS. Perhaps we don't need to split it into two functions?